### PR TITLE
Bookworm Base-image Upgrade

### PR DESCRIPTION
**Pull request by saiarcot895**
#### Why I did it

This PR upgrades SONiC's base image from Debian Bullseye to Debian Bookworm. Bookworm brings in the following notable changes:

* Python is updated from 3.9 to 3.10
* GCC is updated to 12.2.0
* Kernel is updated from 5.10 to 6.1
* Docker is updated from 20.10.14 to 24.0.2
* Various Python package updates

##### Work item tracking
- Microsoft ADO **(number only)**: 25320634

#### How to verify it

KVM tests all pass, basic bootup done on a few platforms to make sure interfaces come up and BGP comes up.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by saiarcot895** [comment]
@xumia @liushilongbuaa FYI, in case of build failures or increased build times

**Comment by liushilongbuaa** [comment]
@saiarcot895 , the HEAD of sonic-utilities is set to your own repo.
9018f3aca3adc959f3f372c04ee62d04241fc244 belongs to your own repo. Please fix it.

diff --git a/src/sonic-utilities b/src/sonic-utilities
index 6833e6cac8c..9018f3aca3a 160000
--- a/src/sonic-utilities
+++ b/src/sonic-utilities
@@ -1 +1 @@
-Subproject commit 6833e6cac8ca0879fde6f5462a994abfa685716d
+Subproject commit 9018f3aca3adc959f3f372c04ee62d04241fc244

File: build_debian.sh [hunk 1]
```diff
@@ -256,24 +256,19 @@ sudo LANG=C chroot $FILESYSTEM_ROOT apt-get -y install apparmor
 sudo cp files/image_config/ntp/ntp-apparmor $FILESYSTEM_ROOT/etc/apparmor.d/local/usr.sbin.ntpd
 sudo LANG=C chroot $FILESYSTEM_ROOT apt-get -y install apt-transport-https \
                                                        ca-certificates \
-                                                       curl \
-                                                       gnupg2 \
```
Comment (qiluo-msft) [inline build_debian.sh:260]: >gnupg2 [](http://example.com/codeflow?start=55&length=6)

why remove gnupg2?

Comment (saiarcot895) [reply build_debian.sh:260]: It was being removed in line 275 below, after docker-ce was installed. This package is technically not needed for the purpose of installing docker-ce.

**Pull request merged**